src/main/java/org/apache/commons/imaging/formats/webp/WebPImageParser.java

```diff
@@ -0,0 +1,326 @@
+import org.apache.commons.imaging.ImagingException;
+import org.apache.commons.imaging.bytesource.ByteSource;
+import org.apache.commons.imaging.common.XmpEmbeddable;
+import org.apache.commons.imaging.common.XmpImagingParameters;
+import org.apache.commons.imaging.formats.tiff.TiffImageMetadata;
+import org.apache.commons.imaging.formats.tiff.TiffImageParser;
+import org.apache.commons.imaging.formats.webp.chunks.WebPChunk;
+import org.apache.commons.imaging.formats.webp.chunks.WebPChunkVP8;
+import org.apache.commons.imaging.formats.webp.chunks.WebPChunkVP8L;
+import org.apache.commons.imaging.formats.webp.chunks.WebPChunkVP8X;
+import org.apache.commons.imaging.formats.webp.chunks.WebPChunkXMP;
+
+import java.awt.Dimension;
+import java.awt.image.BufferedImage;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+
+import static org.apache.commons.imaging.common.BinaryFunctions.read4Bytes;
+import static org.apache.commons.imaging.common.BinaryFunctions.readBytes;
+import static org.apache.commons.imaging.common.BinaryFunctions.skipBytes;
+
+/**
+ * @since 1.0-alpha4
+ */
+public class WebPImageParser extends AbstractImageParser<WebPImagingParameters> implements XmpEmbeddable<WebPImagingParameters> {
+
+    private static final String DEFAULT_EXTENSION = ImageFormats.WEBP.getDefaultExtension();
+    private static final String[] ACCEPTED_EXTENSIONS = ImageFormats.WEBP.getExtensions();
+
+    @Override
+    public WebPImagingParameters getDefaultParameters() {
+        return new WebPImagingParameters();
+    }
+
+    @Override
+    public String getName() {
+        return "WebP-Custom";
+    }
+
+    @Override
+    public String getDefaultExtension() {
+        return DEFAULT_EXTENSION;
+    }
+
+    @Override
+    protected String[] getAcceptedExtensions() {
+        return ACCEPTED_EXTENSIONS;
+    }
+
+    @Override
+    protected ImageFormat[] getAcceptedTypes() {
+        return new ImageFormat[]{ImageFormats.WEBP};
+    }
+
+    /**
+     * Read the file header of WebP file.
+     *
+     * @return file size in file header (including the WebP signature,
+     * excluding the TIFF signature and the file size field).
+     */
+    private static int readFileHeader(InputStream is) throws IOException, ImagingException {
+        byte[] buffer = new byte[4];
+        if (is.read(buffer) < 4 || !WebPConstants.RIFF_SIGNATURE.equals(buffer)) {
+            throw new ImagingException("Not a Valid WebP File");
+        }
+
+        int fileSize = read4Bytes("File Size", is, "Not a Valid WebP File", ByteOrder.LITTLE_ENDIAN);
+        if (fileSize < 0) {
+            throw new ImagingException("File Size is too long:" + fileSize);
+        }
+
+        if (is.read(buffer) < 4 || !WebPConstants.WEBP_SIGNATURE.equals(buffer)) {
+            throw new ImagingException("Not a Valid WebP File");
```

Comment:
Been too long since I worked on the code, but I **think** there is an `ImageReadException`. Have to confirm if other parsers raise that or this error, then try to keep it consistent.

Reply:
No, maybe there was, but it's doing exact same thing as other parsers. Resolving this comment.